crates/storage/store_db/libmdbx.rs

```diff
@@ -152,7 +158,13 @@ impl StoreEngine for Store {
         block_number: BlockNumber,
     ) -> Result<Option<BlockHeader>, StoreError> {
         if let Some(hash) = self.get_block_hash_by_block_number(block_number)? {
-            Ok(self.read_sync::<Headers>(hash.into())?.map(|b| b.to()))
+            match self.read_sync::<Headers>(hash.into())?.map(|b| b.to()) {
+                Some(decoded_header) => match decoded_header {
+                    Ok(decoded_header) => Ok(Some(decoded_header)),
+                    Err(_) => Err(StoreError::Custom("RLP decode failed".to_string())),
+                },
+                None => Ok(None),
+            }
```

Comment:
Multiple match annidation are rarely needed. Here's a refactor suggestion for the whole function:
```rust
fn get_block_header(
    &self,
    block_number: BlockNumber,
) -> Result<Option<BlockHeader>, StoreError> {
    let Some(block_hash) = self.get_block_hash_by_block_number(block_number)? else {
        return Ok(None);
    };

    self.read_sync::<Headers>(block_hash.into())?
        .map(|b| b.to())
        .transpose()
        .map_err(StoreError::from)
}
```